express-mutations/server.ts

```diff
@@ -0,0 +1,105 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/pagila',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+app.use(jsonMiddleware);
+// Endpoint for testing
+app.get('/api/actors/:actorId', async (req, res, next) => {
+  try {
+    const { actorId } = req.params;
+    if (!Number.isInteger(+actorId)) {
+      throw new ClientError(400, `Non-integer actorId: ${actorId}`);
+    }
+    const sql = `
+      select * from "actors"
+      where "actorId" = $1;
+    `;
+    const params = [actorId];
+    const result = await db.query(sql, params);
+    const actor = result.rows[0];
+    if (!actor) throw new ClientError(404, `actor ${actorId} not found`);
+    res.json(actor);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/actors', async (req, res, next) => {
+  try {
+    console.log(req.body);
+    const { firstName, lastName } = req.body;
+    if (!firstName) throw new ClientError(400, 'Enter firstName');
+    if (!lastName) throw new ClientError(400, 'Enter lastName');
```

Comment:
I think a better error message would be `'firstName is required'`. I get what you are saying by doing 'enter firstName', but I don't think it's as clear as saying something is specifically required